controller/src/main/java/io/pravega/controller/server/rpc/auth/PravegaAuthHandler.java

```diff
@@ -0,0 +1,27 @@
+/**
+ * Copyright (c) 2017 Dell Inc., or its subsidiaries. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package io.pravega.controller.server.rpc.auth;
+
+import java.util.Map;
+
+public interface PravegaAuthHandler {
+    enum  PravegaAccessControlEnum {
+        READ,
+        READ_UPDATE
+    }
+
+    String getHandlerName();
+
+    boolean authenticate(Map<String, String> headers);
```

Comment:
This will be taken care of when we implement the token that needs to be sent to segmentstore. When that is implemented the `AuthManager` API will return the token for segmentstore. That will be contained in the `AuthorizedRequest`. Will defer handling of the comment till then. Does this make sense?